cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -573,6 +573,10 @@ func (csr *ClusterStateRegistry) updateReadinessStats(currentTime time.Time) {
 			klog.Warningf("Failed to get nodegroup for %s: %v", unregistered.Node.Name, errNg)
 			continue
 		}
+		if nodeGroup == nil {
```

Comment:
or may be extend the check at L572, `if errNg != nil || nodeGroup == nil {` ?

Reply:
I think it is simpler as two cases; if the check is extended you need an additional check when logging for the case when `ngErr` is `nil`. And semantically they are different. One path is an error, the second is the node simply doesn't exist.